src/dispatch/case/flows.py

```diff
@@ -706,6 +706,14 @@ def common_escalate_flow(
             # Map the case role to an incident role
             incident_role = ParticipantRoleType.map_case_role_to_incident_role(case_roles)
```

Comment:
It seems like the issue is actually here. one can have multiple roles, but this assumes one role. I am also unclear why we pass a list of roles to map and only return one role. Is that expected? It seems like if someone is the reporter, they might get that role here if it is first in the list of case_roles.

Also, the reporter in this case would move to be just a participant.

Reply:
`map_case_role_to_incident_role` is only used here so this seems like an opportunity to improve. Either move the role to a participant (runs the risk of not having an IC if there is no on call service) or If the role comes back as IC, then there can be an additional check if there is already an IC or an IC will be assigned based on an oncall, then remove the IC role from the returned set.